src/BSON/Javascript.c

```diff
@@ -490,17 +487,12 @@ HashTable* php_phongo_javascript_get_properties_hash(zval* object, bool is_debug
 			php_phongo_bson_state state;
 
 			PHONGO_BSON_INIT_STATE(state);
-			if (php_phongo_bson_to_zval_ex(bson_get_data(intern->scope), intern->scope->len, &state)) {
-				Z_ADDREF(state.zchild);
-				zend_hash_str_update(props, "scope", sizeof("scope") - 1, &state.zchild);
-			} else {
-				zval scope;
-
-				ZVAL_NULL(&scope);
-				zend_hash_str_update(props, "scope", sizeof("scope") - 1, &scope);
+			if (!php_phongo_bson_to_zval_ex(bson_get_data(intern->scope), intern->scope->len, &state)) {
+				zval_ptr_dtor(&state.zchild);
+				return NULL;
```

Comment:
`php_phongo_javascript_get_properties_hash` never used to return early. Is this a potential problem if `props` is never freed, or did we previously discuss that PHP itself would clean it up if an exception was thrown during either of the `get_debug_info` or `get_properties` handlers?

Reply:
What I can see from the default `get_debug_info` handler is that it's perfectly fine to return `NULL`: https://github.com/php/php-src/blob/23c65a817390d219bbe77f363cf14956c5c7119b/Zend/zend_object_handlers.c#L152. The function also accepts a `null` return from a `__debuginfo` magic function, so this seems to be ok.

Regardless, I've added a call to `FREE_HASHTABLE` to ensure we free the memory allocated for the hash table when we don't return it.